### PR TITLE
add providers state handling with add and delete functionality

### PR DESCRIPTION
**Pull request by bowenzhu1**
## Notion ticket link

[Add Providers data handling](https://www.notion.so/uwblueprintexecs/134e1f159caa4f73939d11a700c325ad?v=39cf7d5a98384eb9af2bff2265d0ac8a&p=8de6a099d9484095b04f1450fb299ba7&pm=s)

Note: the ticket doesn't have much detail since I kind of impulse did this change, one thing just led to another lol. this is v bad practice 

## Implementation description
* Created ProviderDetails and Providers types
* Added Provider state management within AddChildPage and its relevant children components
* Modified PromptBox to take in name instead of firstName lastName
* Modified PromptBox to take in a delete function
* Implemented add and delete provider functionality through ChildProviderForm

## Steps to test
1. Go to /intake/add-child and the Providers section
2. Try adding providers with the Add Provider button
3. Try deleting providers with the trash icon
4. Try navigating around the Child Information pages to see if data persists locally

## What should reviewers focus on?
* how the Provider state/data is being handled and modified (it's quite messy rn and not great. I'm considering we start using Redux, but I'm seeing if we can manage just passing state around if our app is relatively simple)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/intake/NewProviderModal.tsx [hunk 1]
```diff
@@ -5,13 +5,25 @@ import ModalComponent from "../common/ModalComponent";
 import CustomInput from "../common/CustomInput";
 import OptionalLabel from "./OptionalLabel";
 
+export type ProviderDetails = {
+  providerName: string;
+  providerFileNo: string;
+  primaryPhoneNo: string;
```
Comment (IvanYu327) [inline frontend/src/components/intake/NewProviderModal.tsx:11]: should this also have the type defs for secondary phone number and email and additional comments?

File: frontend/src/components/intake/NewProviderModal.tsx [hunk 2]
```diff
@@ -136,7 +148,17 @@ const NewProviderModal = ({
...
+            primaryPhoneNo,
+            address,
+            relationship,
+          };
+          onClick(newProvider);
+          onClose();
+        }}
         isOpen={isOpen}
         onClose={() => {
           setProviderName("");
```
Comment (IvanYu327) [inline frontend/src/components/intake/NewProviderModal.tsx:164]: related to the above comment, it seems to set the required state variables onClose but doesnt set the secondary phone number or email or additional comments as well.

**Review by IvanYu327 — CHANGES_REQUESTED** [review]
Adding and deleting providers works as expected and they persist when I click around the 3 sections of the intake form. The history.goBack on the edit button caught me off guard when i clicked it before reviewing the code lmao.

File: frontend/src/components/intake/NewProviderModal.tsx (hunk 1 above)
Comment (bowenzhu1) [reply frontend/src/components/intake/NewProviderModal.tsx:11]: oh good catch

**Review by Ezzhingy — APPROVED** [review]
lgtm!
kinda realized now looking back i shouldve just implemented the 'view and edit details btn' without the history.goback()... mb

**Pull request merged**